src/main/kotlin/com/ampnet/payoutservice/controller/PayoutController.kt

```diff
@@ -2,36 +2,55 @@ package com.ampnet.payoutservice.controller
 
 import com.ampnet.payoutservice.controller.request.CreatePayoutRequest
 import com.ampnet.payoutservice.controller.response.CreatePayoutResponse
-import com.ampnet.payoutservice.service.PayoutService
+import com.ampnet.payoutservice.controller.response.CreatePayoutTaskResponse
+import com.ampnet.payoutservice.exception.ErrorCode
+import com.ampnet.payoutservice.exception.ResourceNotFoundException
+import com.ampnet.payoutservice.model.params.CreatePayoutTaskParams
+import com.ampnet.payoutservice.service.CreatePayoutQueueService
 import com.ampnet.payoutservice.util.BlockNumber
 import com.ampnet.payoutservice.util.ChainId
 import com.ampnet.payoutservice.util.ContractAddress
 import com.ampnet.payoutservice.util.WalletAddress
 import org.springframework.http.ResponseEntity
 import org.springframework.security.core.annotation.AuthenticationPrincipal
+import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RestController
+import java.util.UUID
 
 @RestController
-class PayoutController(private val payoutService: PayoutService) {
+class PayoutController(val createPayoutQueueService: CreatePayoutQueueService) {
 
-    @PostMapping("/payout/{chainId}/{assetAddress}/create")
+    @GetMapping("/payouts/tasks/{taskId}")
+    fun getPayoutTaskById(@PathVariable taskId: UUID): ResponseEntity<CreatePayoutTaskResponse> {
+        return createPayoutQueueService.getTaskById(taskId)
+            ?.let { ResponseEntity.ok(it) }
+            ?: throw ResourceNotFoundException(
+                ErrorCode.PAYOUT_TASK_NOT_FOUND,
+                "Create payout task not found"
+            )
+    }
+
+    @PostMapping("/payouts/{chainId}/{assetAddress}/create")
     fun createPayout(
         @PathVariable chainId: Long,
         @PathVariable assetAddress: String,
         @RequestBody requestBody: CreatePayoutRequest,
         @AuthenticationPrincipal requesterAddress: String
     ): ResponseEntity<CreatePayoutResponse> {
-        val response = payoutService.createPayout(
-            chainId = ChainId(chainId),
-            assetAddress = ContractAddress(assetAddress),
-            requesterAddress = WalletAddress(requesterAddress),
-            payoutBlock = BlockNumber(requestBody.payoutBlockNumber),
-            ignoredAssetAddresses = requestBody.ignoredAssetAddresses.mapTo(HashSet()) { WalletAddress(it) }
+        val taskId = createPayoutQueueService.submitTask(
+            CreatePayoutTaskParams(
```

Comment:
https://linear.app/ampnet/issue/SD-556/basic-payout

Proposed create data:
 - name ➖ 
 - a block number that we base our payout on ➕ 
 - an address of token that we base our payout on ❓ - issuer address?
 - a list of wallet addresses to exclude from the payout ➕ 
 - an address of token that we're going to distribute❓- asset address?
 - an amount of token to distribute ➖

Reply:
Sorry for not syncing with you about this, but this is already handled:
- name: stored in blockchain contract (actually it's name or IPFS hash, frontend chooses what it needs there)
- an address of token that we base our payout on - asset address, stored on backend and on blockchain. in this PR issuer is also stored on backend
- an address of token that we're going to distribute - reward token, specified in contract only
- an amount of token to distribute - stored in contract